include/oneapi/dpl/pstl/utils.h

```diff
@@ -879,6 +846,100 @@ struct __is_iterator_type<_T, std::void_t<typename std::iterator_traits<_T>::dif
 template <typename _T>
 static constexpr bool __is_iterator_type_v = __is_iterator_type<_T>::value;
 
+struct __iterators_possibly_equal_impl
+{
+    template <typename _Iterator, typename = std::void_t<>>
+    struct __iterator_value_type
+    {
+        template <typename TIterator>
+        struct unknown_value_type { };
+
+        using __type = unknown_value_type<_Iterator>;
+    };
+
+    template <typename _Iterator>
+    struct __iterator_value_type<_Iterator, std::void_t<std::enable_if_t<std::is_same_v<_Iterator, std::nullptr_t>>>>
+    {
+        using __type = std::nullptr_t;
+    };
+
+    template <typename _Iterator>
+    struct __iterator_value_type<_Iterator, typename std::iterator_traits<_Iterator>::value_type>
+    {
+        using __type = typename std::iterator_traits<_Iterator>::value_type;
+    };
+
+    // Checks that _Iterator1 and _Iterator2 are iterators or pointers,
+    // their iterated values are the same type and the operation of
+    // equality comparison is implemented for them.
+    template <typename _IteratorParam1, typename _IteratorParam2>
+    using __is_equality_comparable =
```

Comment:
I propose to save this formatting because the variant after clang-format looks not so readable like that:
```C++
    template <typename _IteratorParam1, typename _IteratorParam2>
    using __is_equality_comparable =
        std::conjunction<__is_iterator_type<_IteratorParam1>, __is_iterator_type<_IteratorParam2>,
                         std::is_same<typename __iterator_value_type<_IteratorParam1>::__type,
                                      typename __iterator_value_type<_IteratorParam2>::__type>,
                         std::is_invocable<std::equal_to<>, _IteratorParam1, _IteratorParam2>>;
```